include/ppx/application.h

```diff
@@ -321,6 +323,8 @@ class Application
     virtual void DispatchShutdown();
     virtual void DispatchMove(int32_t x, int32_t y);
     virtual void DispatchResize(uint32_t width, uint32_t height);
+    virtual void DispatchWindowIconify(bool iconified);
```

Comment:
Typically, an app should override the function that `Dispatch*` calls. However, apps *can* override the `Dispatch*` functions if there's a need to do it. The `Dispatch*` functions are meant for when someone wants to do something like create their own base Application class because they're going to write a bunch of apps derived from it and they need some custom dispatching handling. Being able to override `Dispatch*` functions preserves the callchain up to the most derived classes without requiring individual applications to process whatever the dispatched is.

For example, lets say someone wants to create framework around BigWheels and they want all applications derived from their application class to do something when a window iconified:
```
class MyBaseApp : public ppx::Application {
protected:
    virtual void DispatchWindowIconify(bool iconified) override {
        if (iconified) PlaySound();
        ppx::Application::DispatchWindowIconify(iconified);
    }
};
```
Every app derived from their `MyBaseApp` class now calls `PlaySound()` when the window is iconfified.

Reply:
Cool, thanks for the explanation!